File: dsa2/tp/tp01/SomaDigitos/main.c
#include<stdio.h>

int sum(char num[], int index){
  if(num[index] == '\0') return 0;
  
  return ((int)(num[index] - '0'))+ sum(num, index+1);
}

int main(){
  char entry[32];
  
  while(scanf("%s", entry) != EOF){
      if(entry[0] == 'F' && entry[1] == 'I' && entry[2] == 'M') break;
      printf("%d\n", sum(entry, 0));
  }
  return 0;
}

File: dsa2/tp/tp01/SomaDigitos/Main.java
import java.util.Scanner;

class Main{
  public static int sum(String num, int index){
    if(index == num.length()) return 0;

    return ((int)(num.charAt(index) - '0')) + sum(num, index+1);
  }

  public static void main(String[] args){
    Scanner sc = new Scanner(System.in);
    String entry;
    
    while(sc.hasNextLine()){
      entry = sc.nextLine();
      if(entry.equals("FIM")) break;
      System.out.println(sum(entry, 0));
    }
    
  }
}
